src/i_system.c

```diff
@@ -111,12 +116,64 @@ static boolean I_ConsoleStdout(void)
 {
 #ifdef _WIN32
     // SDL "helpfully" always redirects stdout to a file.
-    return false;
+    return _isatty(_fileno(stdout));
 #else
     return isatty(fileno(stdout));
 #endif
 }
 
+#ifdef _WIN32
+static void ReopenConsoleHandle(DWORD std, int fd, FILE *stream)
+{
+    HANDLE handle = GetStdHandle(std);
+    DWORD lpmode = 0;
+
+    if (GetConsoleMode(handle, &lpmode))
+    {
+        int unbound_fd = -1;
+
+        FILE *dummy = freopen("CONOUT$", "wt", stream);
```

Comment:
I think the compilers/cppcheck just want the return value of `freopen()` checked for being non-NULL.

Reply:
Can we exclude WIN32 code from cppcheck?  It's only hacks anyway.